design/authenticators/authn_sut/authn_sut_solution_design.md

```diff
@@ -0,0 +1,310 @@
+> machines/applications securely through a proxy machine facilitating
+> streamlined and native workflows for the IT admins. Conjur is now going to
+> integrate with PSM providing integrity across Cyberark products and easy
+> access to the Conjur UI via PSM.
+>
+> In order for PSM to connect to the Conjur UI, a new API(s) would be provided
+> that PSM could call to establish the UI session and login the user directly
+> into the main UI dashboard, without showing the initial login screen. More
+> details will be added in the solution design (TBD).
+>
+> The overall UX will remain the same where users will continue to use the
+> existing workflow of logging onto the PVWA, clicking connect on a defined
+> Conjur instance 1 and connecting through PSM straight into the Conjur UI
+> console.
+
+The SUT authenticator will be the mechanism used to authenticate the user to
+Conjur when they click "Connect" on a Conjur instance in PVWA. PSM will generate
+a SUT for the user using their stored username and password or API key. PSM will
+then pass the SUT to the Conjur UI via a URL parameter. The Conjur UI will use
+the SUT to authenticate the user to Conjur and start a session.
+
+## Assumptions and Prerequisites
+
+Assumptions: Defining the assumptions being made in preparing this Solution
+Design. Detailing why these assumptions are valid and/or acceptable.
+Prerequisites: Do we need additional tooling to develop this feature? Do we need
+additional tooling to test this feature in our pipeline? What needs to be in
+place to build this feature?
+
+## Out-of-scope
+
+- SUTs will not be able to be used to authenticate to Conjur via the CLI
+  - SUTs are intended to be used in automated processes. The CLI is intended to
+    be used by humans (developers, operators, etc.)
+- There are many possible use cases for SUTs. This effort will focus on the
+  basic use case necessary for Conjur UI-PSM integration. Additional use cases
+  may be considered in the future.
+- The PSM integration itself will not be part of this effort. This effort will
+  only focus on the SUT authenticator and the API endpoint for generating SUTs.
+  The PSM team will implement the PSM integration in a separate effort in the
+  PSM codebase.
+
+## Design
+
+### High-Level Design
+
+#### Workflows
+
+##### Basic SUT Workflow
+
+The following diagram shows the basic workflow for using a SUT to authenticate
+to Conjur. The first step is to call the
+`GET /authn-sut/<account>/login` endpoint with the user's
+credentials to generate a SUT. The SUT is then passed to the
+`POST /authn-sut/<account>/authenticate`
+endpoint to authenticate the user to Conjur and retrieve a short-lived auth
+token.
+
+![Basic SUT Workflow](./sut.png)
+
+##### 3rd Party Client Workflow
+
+The following diagram shows the workflow for using a SUT to authenticate to
+Conjur on behalf of a 3rd party client. The first step is to call the
+`GET /authn-sut/<account>/login` endpoint with the user's
```

Comment:
Do we want to have registered client IDs and require the client to sign this request?  We'd have to have a keypair set up to allow the client to request a SUT as part of the configuration, but it would stop the endpoint being used by non-authorized services.